Documentation/ceph-upgrade.md

```diff
@@ -44,6 +44,38 @@ With this upgrade guide, there are a few notes to consider:
   both Rook operator updates and for Ceph version updates.
 * We recommend that you read this document in full before you undertake a Rook cluster upgrade.
 
+## Before you Upgrade
+
+Rook v1.3 has several breaking changes that must be considered **before** upgrading to Rook v1.3.
+1. Ceph Mimic is no longer supported. You must update to Ceph Nautilus **before** updating to Rook v1.3. See the
+   [Ceph upgrade guide](https://rook.github.io/docs/rook/v1.2/ceph-upgrade.html#ceph-version-upgrades).
+2. Directory-based OSDs are no longer supported. If you are specifying a
+   [`directory`](https://github.com/rook/rook/blob/release-1.2/cluster/examples/kubernetes/ceph/cluster-test.yaml#L49-L50)
+   in your CephCluster CR you will need to convert these OSDs to device-based OSDs. See the next section.
+3. OSDs created **before Rook v0.9** are no longer supported during upgrade. To detect if you have these legacy
+   OSDs in your cluster, run `lsblk` on each host to see what partitions exist.
+   - If you see **three partitions** on a device running an OSD, this is a legacy OSD. See below for instructions to convert them.
+   - If you see a single partition or LV with the `ceph` prefix, the OSD is **not** a legacy OSD and you will not need to convert the OSDs.
+
+### Converting Legacy OSDs
+
+If you have legacy OSDs as described in the previous section, it is recommended to migrate the OSDs before the upgrade,
+although the migration is not strictly required before upgrade. If you proceed with the upgrade to v1.3 the OSDs will
+continue running. However, they will no longer be managed by Rook. The OSDs will never again be updated when you update
+Ceph or any other settings in your CephCluster CR.
+
+The procedure to migrate the OSDs requires raw devices or partitions be available as mentioned in the [Ceph Prerequisites](ceph-prerequisites.md).
+After you have available storage devices available in your cluster, do the following:
+1. Configure OSDs on the new devices using the [Storage selection settings](https://rook.github.io/docs/rook/v1.3/ceph-cluster-crd.html#storage-selection-settings).
+   If you have specified `useAllDevices: true`, you may need to restart the operator in order to trigger creation of new OSDs on new devices.
+2. Follow the [OSD Removal Guide](https://rook.github.io/docs/rook/v1.3/ceph-osd-mgmt.html#remove-an-osd) to remove the legacy OSDs.
+   - Before removing each OSD, ensure the PGs are all `active+clean` before continuing with the next OSD to ensure your data is safe.
+
+If you need to re-use your devices and add OSDs to the same devices, you'll need to zap a device before creating a new
+OSD on the same device. For example, `sgdisk --zap-all /dev/<device>` should clean the device and the Rook partitions.
```

Comment:
`sgdisk` is not enough you also need to `dd` at least the first 10MiB of the drive.

Reply:
Instead of mentioning the zap command, i'll add a link to the [cleanup instructions](https://rook.github.io/docs/rook/v1.3/ceph-teardown.html#delete-the-data-on-hosts) to zap the devices. Do we need to update those instructions with `dd`? We have had those same instructions for a long time, so maybe it's not necessary?

Reply:
I'd add a `dd` to the cleanup doc.